packages/plugin-ext/src/main/browser/editor/untitled-resource.ts

```diff
@@ -14,39 +14,91 @@
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  ********************************************************************************/
 
-import { ResourceResolver, Resource } from '@theia/core';
+import { injectable, inject } from 'inversify';
+import { TextDocumentContentChangeEvent } from 'vscode-languageserver-protocol';
+import { Resource, ResourceResolver } from '@theia/core';
 import URI from '@theia/core/lib/common/uri';
-import { injectable } from 'inversify';
 import { Schemes } from '../../../common/uri-components';
+import { FileSystem } from '@theia/filesystem/lib/common/filesystem';
+import { FileSystemWatcher } from '@theia/filesystem/lib/browser/filesystem-watcher';
+import { OpenerService } from '@theia/core/lib/browser';
+import { FileResource } from '@theia/filesystem/lib/browser';
 
 const resources = new Map<string, UntitledResource>();
```

Comment:
It would be good to convert such variables to properties of `UntitledResourceResolver`.

Reply:
It should be a property in `UntitledResourceResolver`

Reply:
done